packages/nuxt/src/core/nuxt.ts

```diff
@@ -122,10 +122,28 @@ async function initNuxt (nuxt: Nuxt) {
 
   // Init user modules
   await nuxt.callHook('modules:before')
-  const modulesToInstall = [
-    ...nuxt.options.modules,
-    ...nuxt.options._modules
-  ]
+  const modulesToInstall = []
+
+  const watchedPaths = new Set<string>()
+
+  // Automatically register user modules
+  for (const config of nuxt.options._layers.map(layer => layer.config).reverse()) {
+    const layerModules = await resolveFiles(config.srcDir, [
+      `${config.dir?.modules || 'modules'}/*{${nuxt.options.extensions.join(',')}}`,
+      `${config.dir?.modules || 'modules'}/*/index{${nuxt.options.extensions.join(',')}}`
+    ])
+    for (const mod of layerModules) {
+      modulesToInstall.push(mod)
```

Comment:
I belive we still need to deduplicate and avoid pushing if already exist in either `modulesToInstall` or `nuxt.options.modules` them here before pushing this to the module installer to try and deduplicate.

Reply:
yes, we _now_ do, because we are adding the explicit modules last.

Reply:
pushed a fix - wdyt?

Reply:
Looks good!